src/change_stream/mod.rs

```diff
@@ -0,0 +1,172 @@
+/// "resumable" errors, such as transient network failures. It can also be done manually by passing
+/// a ResumeToken retrieved from a past event into either the resume_after or start_after (4.2+)
+/// options used to create the ChangeStream. Issuing a raw change stream aggregation is discouraged
+/// unless users wish to explicitly opt out of resumability.
+///
+/// A `ChangeStream` can be iterated to return batches of instances of any type that implements the
+/// Deserialize trait. By default, it returns batches of `ChangeStreamEventDocument`. These
+/// documents correspond to changes in the associated collection, database or deployment.
+///
+/// A change stream can be iterated like any other `Iterator`:
+///
+/// ```rust
+/// #[cfg(not(feature = "sync"))]
+/// # use futures::stream::StreamExt;
+/// # use mongodb::{Client, error::Result, bson::doc,
+/// # change_stream::document::ChangeStreamEventDocument};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #
+/// # #[cfg(not(feature = "sync"))]
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// let result = coll.insert_one(doc! { "x": 1 }, None).await?;
+/// while let Some(change) = change_stream.next().await {
+///     let document: ChangeStreamEventDocument = change?;
+///     let full_document =
+///     println!("operation performed: {:?}, document: {:?}", document.operation_type, document.full_document);
+///     // operation performed: Insert, document: { "x": 1 }
+/// }
+/// #
+/// # Ok(())
+/// # }
+/// ```
+///
+/// A ChangeStream allows any Deserialize type for `ChangeStreamEventDocument` to be specified:
+/// ```rust
+/// #[cfg(not(feature = "sync"))]
+/// # use bson::Document;
+/// # use futures::stream::StreamExt;
+/// # use serde::Deserialize;
+/// # use mongodb::{Client, error::Result, bson::doc, change_stream::document::ChangeStreamEventDocument};
+/// # #[cfg(feature = "async-std-runtime")]
+/// # use async_std::task;
+/// # #[cfg(feature = "tokio-runtime")]
+/// # use tokio::task;
+/// #[derive(Deserialize, Debug)]
+/// pub struct UserDocument {
+///     pub full_document: Option<Document>,
+/// }
+/// #
+/// # #[cfg(not(feature = "sync"))]
+/// # async fn func() -> Result<()> {
+/// # let client = Client::with_uri_str("mongodb://example.com").await?;
+/// # let coll = client.database("foo").collection("bar");
+/// let mut change_stream = coll.watch(None, None).await?;
+/// let result = coll.insert_one(doc! { "x": 1 }, None).await?;
+
+/// while let Some(change) = change_stream.next().await {
+///     let document: UserDocument = change?;
+///     println!("document: {:?}", document.full_document)     
+///     // document: { "x": 1 }
```

Comment:
ditto showing what this would actually look like

Reply:
fixed